src/flint/fmpz.pyx

```diff
@@ -298,6 +378,21 @@ cdef class fmpz(flint_scalar):
         if stype == FMPZ_TMP: fmpz_clear(sval)
         return u
 
+    def __rpow__(s, t, m):
+        cdef fmpz_struct tval[1]
+        cdef int stype = FMPZ_UNKNOWN
+        cdef ulong exp
+        u = NotImplemented
+        if m is not None:
+            raise NotImplementedError("modular exponentiation")
+        ttype = fmpz_set_any_ref(tval, t)
+        if ttype != FMPZ_UNKNOWN:
+            u = fmpz.__new__(fmpz)
+            s_ulong = fmpz_get_ui(s.val)
+            fmpz_pow_ui((<fmpz>u).val, tval, s_ulong)
```

Comment:
I'm not sure about this.

Actually I see that with current master and Cython 0.29 we have:
```
>>> import flint
>>> flint.fmpz(2) ** flint.fmpz(-2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "src/flint/fmpz.pyx", line 297, in flint._flint.fmpz.__pow__
    fmpz_pow_ui((<fmpz>u).val, sval, c)
OverflowError: can't convert negative value to ulong
```
So I guess the existing `__pow__` method should be changed. At least `OverflowError` is not the right exception to raise.

Python itself gives:
```python
>>> 2 ** -2
0.25
>>> 1 ** -1
1.0
>>> 2 ** 2
4
```
In principle an `fmpq` could be returned but I'm not sure if that's better than an error.

Reply:
gmpy2 returns `mpfr`:
```python
In [3]: gmpy2.mpz(2) ** -2
Out[3]: mpfr('0.25')

In [4]: gmpy2.mpz(2) ** 2
Out[4]: mpz(4)
```
It's probably good to maintain compatibility with `gmpy2.mpz` and `int` itself if possible so that `mpz` can be used interchangeably with those. I'm not sure what type that really means in the `python-flint` context though.